src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/LegacyCSProjPackageReferenceProject.cs

```diff
@@ -278,26 +281,35 @@ private PackageSpec GetPackageSpec()
 
             // In legacy CSProj, we only have one target framework per project
             var tfis = new TargetFrameworkInformation[] { projectTfi };
-            return new PackageSpec(tfis)
+
+            return new PackageSpec()
             {
                 Name = _projectName ?? _projectUniqueName,
+                Version = new NuGetVersion(1, 0, 0, 0, (string)null, string.Empty),
```

Comment:
This should read the Version property from the project file. ProjectVersion may override it. This all needs to match pack behavior.